engine/runtime/src/main/java/org/enso/interpreter/runtime/error/Warning.java

```diff
@@ -85,6 +85,25 @@ public static WithWarnings attach(EnsoContext ctx, Object value, Object warning,
     return new WithWarnings(value, new Warning(warning, origin, ctx.clockTick()));
   }
 
+  @Builtin.Method(
+      name = "has_warnings_builtin",
+      description = "Are any warnings attached to the value.",
+      autoRegister = false)
+  @Builtin.Specialize
+  @CompilerDirectives.TruffleBoundary
```

Comment:
Placing this check behind `@TruffleBoundary` will slow the check down significantly.